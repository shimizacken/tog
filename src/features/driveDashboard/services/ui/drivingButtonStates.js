export const DrivingButtonStates = {
    TURNED_OFF: 'TURNED_OFF',
    READY: 'READY',
    DRIVE: 'DRIVE',
    ERROR: 'ERROR'
};