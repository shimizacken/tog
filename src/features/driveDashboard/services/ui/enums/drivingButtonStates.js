const DrivingButtonStates = {
    TURNED_OFF: 'TURNED_OFF',
    READY: 'READY',
    DRIVE: 'DRIVE',
    ERROR: 'ERROR'
};

export default DrivingButtonStates;
