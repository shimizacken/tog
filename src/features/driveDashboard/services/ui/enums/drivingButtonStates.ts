export enum DrivingButtonStates {
  TURNED_OFF = "TURNED_OFF",
  READY = "READY",
  DRIVE = "DRIVE",
  ERROR = "ERROR",
  SWITCHING = "SWITCHING",
}
