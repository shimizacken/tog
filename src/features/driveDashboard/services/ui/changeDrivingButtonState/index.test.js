import calculateDrivingButtonState from './index';
import { DrivingButtonStates } from '../drivingButtonStates';

describe('calculateDrivingButtonState test suite', () => {
  
    describe('default state', () => {
        
        it('should return default state', () => {
            
            expect(calculateDrivingButtonState()).toMatchSnapshot();
        });    
    });
    
    describe('rest states', () => {
      
        it('should return READY when TURNED_OFF', () => {
        
            const currentState = DrivingButtonStates.TURNED_OFF;
            
            expect(calculateDrivingButtonState(currentState)).toMatchSnapshot();
        });

        it('should return DRIVE when READY', () => {
        
            const currentState = DrivingButtonStates.READY;
            
            expect(calculateDrivingButtonState(currentState)).toMatchSnapshot();
        });

        it('should return READY when DRIVE', () => {
        
            const currentState = DrivingButtonStates.DRIVE;
            
            expect(calculateDrivingButtonState(currentState)).toMatchSnapshot();
        });
    });
    
});
