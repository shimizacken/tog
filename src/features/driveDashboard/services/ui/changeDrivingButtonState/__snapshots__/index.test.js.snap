// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateDrivingButtonState test suite default state should return default state 1`] = `
Object {
  "color": "ORANGE",
  "drivingButtonStates": "READY",
  "seconds": 2,
}
`;

exports[`calculateDrivingButtonState test suite rest states should return DRIVE when READY 1`] = `
Object {
  "color": "GREEN",
  "drivingButtonStates": "DRIVE",
  "seconds": 2,
}
`;

exports[`calculateDrivingButtonState test suite rest states should return READY when DRIVE 1`] = `
Object {
  "color": "ORANGE",
  "drivingButtonStates": "READY",
  "seconds": 2,
}
`;

exports[`calculateDrivingButtonState test suite rest states should return READY when TURNED_OFF 1`] = `
Object {
  "color": "ORANGE",
  "drivingButtonStates": "READY",
  "seconds": 2,
}
`;
